src/dsp_tools/commands/xml_validate/prepare_input.py

```diff
@@ -0,0 +1,150 @@
+    return ProjectDeserialised(
+        shortcode=project.shortcode, default_onto=project.default_onto, permissions=permissions, resources=resources
+    )
+
+
+def _deserialise_one_permission(permission: PermissionsXML) -> PermissionsDeserialised:
+    permission_dict = {x.attrib["group"]: x.text for x in permission.permission_eles}
+    permissions = cast(dict[str, str], permission_dict)
+    return PermissionsDeserialised(permission_id=permission.permission_id, permission_dict=permissions)
+
+
+def _deserialise_one_resource(resource: ResourceXML) -> ResourceDeserialised:
+    res_id = resource.res_attrib["id"]
+    values: list[ValueDeserialised] = []
+    for val in resource.values:
+        values.extend(_deserialise_one_property(val))
+    return ResourceDeserialised(
+        res_id=res_id,
+        res_class=resource.res_attrib["restype"],
+        label=resource.res_attrib["label"],
+        permissions=resource.res_attrib.get("permissions"),
+        values=values,
+    )
+
+
+def _deserialise_one_property(prop_ele: etree._Element) -> list[ValueDeserialised]:
+    match prop_ele.tag:
+        case "text-prop":
+            return _deserialise_text_prop(prop_ele)
+        case "list-prop":
+            return _deserialise_list_prop(prop_ele)
+        case "resptr-prop":
+            return _deserialise_resptr_prop(prop_ele)
+        case _:
+            return []
+
+
+def _deserialise_text_prop(prop: etree._Element) -> list[ValueDeserialised]:
+    prop_name = prop.attrib["name"]
+    all_vals: list[ValueDeserialised] = []
+    for child in prop.iterchildren():
+        permissions = child.attrib.get("permissions")
+        comments = child.attrib.get("comment")
+        val = cast(str, child.text)
+        match child.attrib["encoding"]:
+            case "utf8":
+                all_vals.append(
+                    SimpleTextDeserialised(
+                        prop_name=prop_name, prop_value=val, permissions=permissions, comments=comments
+                    )
+                )
+            case "xml":
+                all_vals.append(
+                    RichtextDeserialised(
+                        prop_name=prop_name, prop_value=val, permissions=permissions, comments=comments
+                    )
+                )
+    return all_vals
+
+
+def _deserialise_list_prop(prop: etree._Element) -> list[ValueDeserialised]:
+    prop_name = prop.attrib["name"]
+    list_name = prop.attrib["list"]
+    all_vals: list[ValueDeserialised] = []
+    for val in prop.iterchildren():
+        txt = cast(str, val.text)
+        all_vals.append(
+            ListValueDeserialised(
+                prop_name=prop_name,
+                prop_value=txt,
+                list_name=list_name,
+                permissions=val.attrib.get("permissions"),
+                comments=val.attrib.get("comment"),
+            )
+        )
+    return all_vals
+
+
+def _deserialise_resptr_prop(prop: etree._Element) -> list[ValueDeserialised]:
```

Comment:
I would be very happy if we could stop using the term `resptr` in any new code we write. It's a "Link (Value)" in our domain language, and code should follow domain terminology

Reply:
I agree in general and disagree in this specific case.
The reason why I included `resptr` in this function name is because it takes an element with that tag. This XML syntax is being removed here and the output corresponds to the domain language. This is what I wanted to reflect in the function name. But if you feel strongly about it I can rename it.